packages/messaging/ios/RNFBMessaging/RNFBMessaging+UNUserNotificationCenter.m

```diff
@@ -53,16 +53,18 @@ - (nullable NSDictionary *)getInitialNotification {
 }
 
 - (void)userNotificationCenter:(UNUserNotificationCenter *)center willPresentNotification:(UNNotification *)notification withCompletionHandler:(void (^)(UNNotificationPresentationOptions options))completionHandler {
-  NSDictionary *notificationDict = [RNFBMessagingSerializer notificationToDict:notification];
+  if (notification.request.content.userInfo[@"gcm.message_id"]) {
+    NSDictionary *notificationDict = [RNFBMessagingSerializer notificationToDict:notification];
 
-  // Don't send an event if contentAvailable is true - application:didReceiveRemoteNotification will send the event
-  // for us, we don't want to duplicate them
-  if (!notificationDict[@"contentAvailable"]) {
-    [[RNFBRCTEventEmitter shared] sendEventWithName:@"messaging_message_received" body:notificationDict];
-  }
+    // Don't send an event if contentAvailable is true - application:didReceiveRemoteNotification will send the event
+    // for us, we don't want to duplicate them
+    if (!notificationDict[@"contentAvailable"]) {
+      [[RNFBRCTEventEmitter shared] sendEventWithName:@"messaging_message_received" body:notificationDict];
+    }
 
-  // TODO in a later version allow customising completion options in JS code
-  completionHandler(UNNotificationPresentationOptionNone);
+    // TODO in a later version allow customising completion options in JS code
+    completionHandler(UNNotificationPresentationOptionNone);
```

Comment:
Shouldn't this be outside of the if? not sure...

Reply:
We don't want to complete notifications that are presented from other libraries, if we complete them for other libraries with `UNNotificationPresentationOptionNone` then they won't show in the foreground if that is the users expected behaviour from other third party libs

Reply:
I'd imagine other libs would have their own `willPresentNotification` setup hopefully

Reply:
@Salakar Thank you for this pr! It's really helpful. I just have one question. For
```- (void)userNotificationCenter:(UNUserNotificationCenter *)center didReceiveNotificationResponse:(UNNotificationResponse *)response withCompletionHandler:(void (^)(void))completionHandler {
  NSDictionary *remoteNotification = response.notification.request.content.userInfo;
  if (remoteNotification[@"gcm.message_id"]) {
    NSDictionary *notificationDict = [RNFBMessagingSerializer remoteMessageUserInfoToDict:remoteNotification];
    [[RNFBRCTEventEmitter shared] sendEventWithName:@"messaging_notification_opened" body:notificationDict];
    _initialNotification = notificationDict;
  }
  completionHandler();
}
```
why, in this case, completionHandler is outside the if check?